### PR TITLE
Remove deprecated reference to Buffer, unused function

### LINKED ISSUE #167: Adjust use of Buffer - deprecation warning
Got this warning:
```
(node:9846) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
```

Found buffer [where we're converting to base64 in scope.js](https://github.com/plocket/docassemble-cucumber/blob/main/lib/scope.js#L226).

### PR DESCRIPTION
**Pull request by niharikasingh**
Closes #167 

The ticket asked for the `Buffer` usage to be rewritten, but we have a nice new function that works properly so I think we can just use that instead. Let me know if that's incorrect.

### CODE REVIEW

File: lib/scope.js [hunk 1]
```diff
@@ -608,9 +608,15 @@ module.exports = {
...
+  toBase64: async function ( scope, { utf8_str }, exclude_padding=false) {
+    // Sometimes, converting to base64 results in padding characters 
+    // being added to the end. Docassemble does not use these padding
+    // characters everywhere, so we allow for stripping them using 
+    // the exclude_padding argument. For more details, see
+    // https://github.com/nodejs/node/issues/6107#issuecomment-207177131
     let buffer = Buffer.from( utf8_str, 'utf-8' );
-    return buffer.toString( `base64` );
+    let base64str = buffer.toString( `base64` )
+    return exclude_padding ? base64str.replace(/[^A-Za-z0-9]/g, '') : base64str;
```
Comment (plocket) [inline lib/scope.js:619]: Does docassemble use these padding characters anywhere? Maybe the default here should be `true`?

Comment (niharikasingh) [reply lib/scope.js:619]: works for me. i wasn't sure if docassemble ever kept the padding characters around.

**Comment by niharikasingh** [comment]
Line 321 deserves special attention. Is `encoded_trigger_var = encoded_trigger_var || await scope.toBase64( scope, { utf8_str: `None` });` correct? Or should padding me included there?

**Comment by plocket** [comment]
> Is encoded_trigger_var = encoded_trigger_var || await scope.toBase64( scope, { utf8_str: None }); correct?

It honestly doesn't matter if it's included or not as it's a dummy value. Maybe I should make that more clear, so we can make that an issue.

Unrelatedly, I think I should have deleted `encoded_trigger_var ||` considering this can only happen `if ( !encoded_trigger_var )`, so that's good to know too.

File: lib/scope.js (hunk 1 above)
Comment (plocket) [reply lib/scope.js:619]: Sorry I didn't catch this before. It occurred to me that I may have seen characters outside of those in base64 strings in the past: https://stackoverflow.com/a/13195218/14144258. I think the safest thing to do might be to just remove any `=` in there? That would probably be `/=/g`.

**Review by plocket — APPROVED** [review]
Thank you! It'll be nice not to get that deprecation warning on every single run lol

**Pull request merged**